components/count-badge/count-badge.js

```diff
@@ -134,7 +172,15 @@ class CountBadge extends RtlMixin(LitElement) {
 			numberString = `${'9'.repeat(this.maxDigits)}+`;
 		}
 		return html`
-        	<div class="d2l-count-badge-number">${numberString}</div>`;
+        	<div
+			id="${this._badgeId}"
+			tabindex="${ifDefined(this.tabStop || this.tooltip ? '0' : undefined)}" 
+			role="${ifDefined(this.announceChanges ? 'status' : undefined)}"
+			aria-label="${this.text}">
+				<div class="d2l-count-badge-number" aria-hidden="true">${numberString}</div>
+			${this.tooltip ? html`<d2l-tooltip for="${this._badgeId}">${this.text}</d2l-tooltip>` : html`<span class="d2l-offscreen">"${this.text}"</span>`}
```

Comment:
We only need one of the tooltip or offscreen so I just have a ternary check here to show the tooltip if required, otherwise use d2l-offscreen.